core/citrus-base/src/main/java/com/consol/citrus/functions/core/ReadFileResourceFunction.java

```diff
@@ -49,12 +50,23 @@ public String execute(List<String> parameterList, TestContext context) {
 
         try {
             if (base64) {
-                return Base64.encodeBase64String(FileCopyUtils.copyToByteArray(FileUtils.getFileResource(parameterList.get(0), context).getInputStream()));
+                return Base64.encodeBase64String(readFileContent(parameterList, context).getBytes(FileUtils.getCharset(parameterList.get(0))));
             } else {
-                return context.replaceDynamicContentInString(FileUtils.readToString(FileUtils.getFileResource(parameterList.get(0), context), FileUtils.getCharset(parameterList.get(0))));
+                return readFileContent(parameterList, context);
             }
         } catch (IOException e) {
             throw new CitrusRuntimeException("Failed to read file", e);
         }
     }
+
+    /**
+     * Read the file content replacing dynamic content in the file content
+     * @param parameterList
+     * @param context
+     * @return
+     * @throws IOException
+     */
+    private String readFileContent(List<String> parameterList, TestContext context) throws IOException {
```

Comment:
How about using a method argument `filePath` instead of the whole parameter list? Only the first entry in the list (the file path) is used in the method so no reason to pass in the whole list as an argument.